ls1intum/Artemis
# `Communication`: Remove trailing hyphens on auto-generated channel names

### Checklist
#### General


#### Server
- [x] I added multiple integration tests (Spring) related to the features (with a high test coverage).(https://docs.artemis.cit.tum.de/dev/guidelines/server/#rest-endpoint-best-practices-for-authorization) and checked the course groups for all new REST Calls (security).
#### Client

### Motivation and Context


With #7086 auto-generated channel names do not contain duplicate hyphens anymore. However, it could still happen, that a auto-generated channel name ends with a hyphen.
### Description

Trailing hyphens get removed when the input field for the channel name gets initialized or whenever the user enters a title with trailing hyphens (or special characters).
The server also removes trailing hyphens, whenever it generates a channel name based on the lecture/exercise/exam title.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Course with messaging enabled

1. Create a new lecture with a name that ends with special characters
2. Confirm that the auto-typed channel name does not end with a hyphen
3. Save the lecture
4. Import the newly created lecture
5. Confirm that the auto-generated channel name does not end with a hyphen (edit the lecture or look at the messages tab)

### Review Progress






#### Performance Review
#### Code Review
#### Manual Tests

### Test Coverage






#### Client

| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|---------------------:|
| [title-channel-name.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5347/latest/artifact/shared/Coverage-Report-Client-Tests/app/shared/form/title-channel-name/title-channel-name.component.ts.html) | 100% | ✅ |

#### Server

| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|---------------------:|
| [ChannelService.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5347/latest/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.service.metis.conversation/ChannelService.html) | 91% | ✅ |

### Screenshots


before:
<img width="958" alt="before" src="https://github.com/ls1intum/Artemis/assets/44754405/c91766c8-d2c2-4112-82d7-137da6c7226b">

after:
<img width="958" alt="after" src="https://github.com/ls1intum/Artemis/assets/44754405/9f4db2be-fc32-4f83-a56f-c351ffb56b6b">
